modules/axhal/src/platform/x86_pc/rtc.rs

```diff
@@ -0,0 +1,381 @@
+            8 => secs += 18_316_800,
+            9 => secs += 20_995_200,
+            10 => secs += 23_587_200,
+            11 => secs += 26_265_600,
+            12 => secs += 28_857_600,
+            _ => (),
+        }
+
+        secs += (day as u64 - 1) * 86_400;
+        secs += hour as u64 * 3600;
+        secs += minute as u64 * 60;
+        secs += second as u64;
+
+        secs
+    }
+
+    /// Get time
+    pub fn time(&mut self) -> u64 {
+        loop {
+            unsafe {
+                self.wait(false);
+                let time = self.time_no_wait();
+                self.wait(false);
+                let next_time = self.time_no_wait();
+                if time == next_time {
+                    return time;
+                }
+            }
+        }
+    }
+
+    pub unsafe fn write_time_no_wait(&mut self, unix_time: u32) {
+        let second = self.read(0) as usize;
+        let minute = self.read(2) as usize;
+        let hour = self.read(4) as usize;
+        let day = self.read(7) as usize;
+        let month = self.read(8) as usize;
+        let year = self.read(9) as usize;
+        let _century = 20;
+        let register_b = self.read(0xB);
+
+        debug!(
+            "{}, {}, {}, {}, {}, {}, {}, {}",
+            second,
+            minute,
+            hour,
+            day,
+            month,
+            year,
+            register_b,
+            register_b & 4
+        );
+
+        let secs = unix_time;
+
+        // Calculate date and time
+        let t = secs;
+        let mut tdiv = t / 86400;
+        let mut tt = t % 86400;
+        let mut hour = tt / 3600;
+        tt %= 3600;
+        let mut min = tt / 60;
+        tt %= 60;
+        let mut sec = tt;
+        let mut year = 1970;
+        let mut mon = 1;
+
+        // 计算年、月和日
+        while tdiv >= 365 {
+            let days = if is_leap_year(year) { 366 } else { 365 };
+            if tdiv >= days {
+                tdiv -= days;
+                year += 1;
+            } else {
+                break;
+            }
+        }
+
+        while tdiv > 0 {
+            let days = days_in_month(mon, year);
+            if u64::from(tdiv) >= days {
+                tdiv -= days as u32;
+                mon += 1;
+            } else {
+                break;
+            }
+        }
+
+        let mut mday = tdiv + 1;
+
+        year -= 2000;
+
+        if register_b & 4 != 4 {
+            sec = cvt_dec(sec as usize) as u32;
+            min = cvt_dec(min as usize) as u32;
+            mday = cvt_dec(mday as usize) as u32;
+            mon = cvt_dec(mon as usize) as u64;
+            year = cvt_dec(year as usize) as u64;
+        }
+        let mut bcd_value = hour % 10;
+        let tens = hour / 10;
+        if hour >= 12 {
+            bcd_value |= 0x80;
+        }
+        bcd_value |= tens << 4;
+        hour = bcd_value;
+
+        debug!("{}, {}, {}, {}, {}, {}", sec, min, hour, mday, mon, year);
```

Comment:
This debug message should be more precise: <which function/action/...> <which field> : <value> ..... Or just remove this.